WalletWasabi.Fluent/ViewModels/MainViewModel.cs

```diff
@@ -194,7 +196,26 @@ private void RegisterViewModels()
 					return null;
 				});
 
-			LegalDocumentsViewModel.RegisterLazy(() => new LegalDocumentsViewModel(_global.LegalChecker));
+			RxApp.MainThreadScheduler.Schedule(async () =>
```

Comment:
There are two events in LegalChecker: 

- ProvisionalChanged
- AgreedChanged

They can be used instead of a possibly endless loop.